### DISCUSSION TITLE
is it me, or ...

### DISCUSSION THREAD

**Discussion by Sorcerer13** [pybricks/support#1301] [Q&A] [answered]
I've had consistent problems with Pybricks on the Edge browser. After a (short) while on Pybricks, my internet connection (via Plusnet) drops, and I have inordinate problems getting it back again, with messages on my PC saying - no internet/internet not responding/internet becoming available shortly/internet unavailable. Even when I have a connection it becomes unstable after using Pybricks in a browser.
 I've seen references to a 'local' Pybricks app, but finding it seems very elusive - can anyone help? Does it still exist?

**Comment by BertLindeman** [comment]
On the configuration tab on the left [edit] near the bottom:
![afbeelding](https://github.com/pybricks/support/assets/8142081/daa3c42f-62ce-43cb-9fa4-48d989b52777)

**Comment by Sorcerer13** [comment]
Erm - that option isn't present for me! (see attached)
Im running v3.3.0 (Pybricks code v2.2.0).
It does say "ready for offline use", but it says Daz on buses, but they
don't wash your clothes!

**Reply by BertLindeman** [reply]
Attachment not in github. Can you insert the picture here?
The beta.pybricks,com and code.pybricks.com are (AT the moment) equal.

**Comment by Sorcerer13** [comment]
Picture is here (hopefully)
![MissingLocalApp](https://github.com/pybricks/support/assets/33292274/94615b12-a3c2-4228-8c3b-137cf44252f4)

**Reply by BertLindeman** [reply]
Picture **is** there. 

It's really outside my competence, 
but you could try [ctrl]-F5 to do a refresh.
The refresh can take some time to complete.
It might help.

**Reply by Sorcerer13** [reply]
Thanks for trying Bert - it's all one can ask for!
I'll try the refresh, then try reinstalling everything, which could take a while - I was just hoping for a quick "You idiot, press F5 whilst sticking a finger in your ear" solution...

**Reply by BertLindeman** [reply]
Or try GoogleChrome?

Good luck.

**Comment by Sorcerer13** [comment, marked as answer]
Just tried getting hold of Chrome, and the "App" option is present!
It must be another 'benefit' of Microsoft's Edge.
It looks like using Chrome I can install it as an app, so that shoul hopefully solve the issue (or at least get me a shade closer to Nirvana!).
Thanks again Bert for giving up your time to help an old fool!

**Reply by BertLindeman** [reply]
A fool would not ask for a bit of help! 😃

**Comment by dlech** [comment]
Based on the screenshot, it looks like Pybricks was already installed as an app using the Edge browser. It works like this:

If Pybricks Code is *not* installed as an app, you will see the "Install as App" button and there will also be an icon in the address bar provided by the browser that does the same thing.

![image](https://github.com/pybricks/support/assets/963645/ec5518c5-7ab7-4529-ac2b-4efa56be0bf3)

![image](https://github.com/pybricks/support/assets/963645/8e483a9b-8bc8-4ac3-8837-d335b0a7121d)

Once you have "installed" it (which really just means add a desktop shortcut, there is not a separate "app" version), the button will disappear from Pybricks Code and the icon in the address bar will change.

![image](https://github.com/pybricks/support/assets/963645/18c77c83-5c48-4bbf-a72d-85e64f6bcb8a)

This icon matches the one seen in https://github.com/orgs/pybricks/discussions/1301#discussioncomment-7666237, so clearly, it was already installed as an app.

Unfortunately, there are technical limitations that make it difficult to provide a better user experience in this regard. Discussed in https://github.com/pybricks/pybricks-code/issues/1161.

**Reply by dlech** [reply]
Another way to see what is installed as an app is through the main menu in the browser.

![image](https://github.com/pybricks/support/assets/963645/116266fe-5950-4475-a357-a7caa354b87f)
![image](https://github.com/pybricks/support/assets/963645/2a7b721e-3110-48f9-970e-241654ecf5b7)

**Comment by dlech** [comment]
> I've seen references to a 'local' Pybricks app, but finding it seems very elusive - can anyone help? Does it still exist?

I mentioned this in one of my other comments, but to make it easier to find, I will mention it again.

"Install as App" only creates shortcuts to load the web page in a special window that looks a bit more like a regular app but is still really a web browser with the address bar and other things hidden. There is no separate "app" version of Pybricks. It is a [Progressive Web App](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps) so the web page and the "app" are literally the same thing. And both work offline as long as it says "Ready for offline use.". You don't have to "install as app" to make it work offline. Installing as an app and working offline are completely unrelated features.

**Comment by dlech** [comment]
> After a (short) while on Pybricks, my internet connection (via Plusnet) drops, and I have inordinate problems getting it back again, with messages on my PC saying - no internet/internet not responding/internet becoming available shortly/internet unavailable. Even when I have a connection it becomes unstable after using Pybricks in a browser.

If Pybricks Code really is triggering the Internet connection issue and it is not just a coincidence that this happened at this time, then I would suspect that the computer has a dual Wi-Fi/Bluetooth chip that was not well designed and trying to use both Bluetooth and Wi-Fi at the same time is causing the problem.
